dbt_semantic_interfaces/validations/dimension_const.py

```diff
@@ -79,10 +79,8 @@ def _validate_dimension(
             if dimension.reference not in time_dims_to_granularity and dimension.type_params:
                 time_dims_to_granularity[dimension.reference] = dimension.type_params.time_granularity
 
-                # The primary time dimension can be of different time granularities, so don't check for it.
                 if (
                     dimension.type_params is not None
-                    and not dimension.type_params.is_primary
```

Comment:
Oh huh. This is vestigial from the time before `metric_time`, right?

Reply:
That is correct.